File: src/main/java/cz/grunwjir/csasdemo/csas/client/CsasResponse.java
package cz.grunwjir.csasdemo.csas.client;

import lombok.Data;

import java.util.List;

/**
 * The wrapper class for data responses including paging information.
 *
 * @param <T> data content
 * @author Jiri Grunwald
 */
@Data
public class CsasResponse<T> {

    private Integer pageNumber;
    private Integer pageCount;
    private Integer pageSize;
    private Integer totalItemCount;
    private List<T> items;
}

File: src/main/kotlin/cz/grunwjir/csasdemo/csas/client/CsasResponse.kt
package cz.grunwjir.csasdemo.csas.client

/**
 * The wrapper class for data responses including paging information.
 *
 * @param <T> data content
 * @author Jiri Grunwald
 */
data class CsasResponse<T>(
    val pageNumber: Int,
    val pageCount: Int,
    val pageSize: Int,
    val totalItemCount: Int,
    val items: List<T>
)